### ISSUE TITLE
Index on Reference Column break Schema Generation

### ISSUE THREAD

**Issue by maphe** [mikro-orm/mikro-orm#694]
**Describe the bug**
I'm trying to generate a sqlite schema with an index on 2 columns, one of which being a relation. The generator fails as the index is attempted to be created before the reference column.

**Stack trace**
```
Error: create index `index_name_and_parent` on `children` (`name`, `parent_id`); - SQLITE_ERROR: no such column: parent_id
```

**To Reproduce**
Parent Entity
```ts
import { Collection, Entity, OneToMany, PrimaryKey } from 'mikro-orm';
import { Child }                                     from './Child';

@Entity({ tableName: 'parents' })
export class Parent {
  @PrimaryKey()
  id!: number;

  @OneToMany({ entity: () => Child, mappedBy: 'parent' })
  children = new Collection<Child>(this);
}
```

Child Entity
```ts
import { Entity, Index, ManyToOne, PrimaryKey, Property } from 'mikro-orm';
import { Parent }                                         from './Parent';

@Entity({ tableName: 'children' })
@Index({ name: 'index_name_and_parent', properties: ['name', 'parent'] })
export class Child {
  @PrimaryKey()
  id!: number;

  @Property()
  name: string;

  @ManyToOne({ entity: () => Parent, nullable: true })
  parent: Parent;
}
```

Schema Generation:
```ts
    const orm = await MikroORM.init({
      entities: [Parent, Child],
      name: 'sfp_test',
      dbName: 'sfp_db_test',
      type: 'sqlite',
      discovery: { disableDynamicFileAccess: true },
      cache: { enabled: false },
    });
    const generator = orm.getSchemaGenerator();
    console.log(await generator.generate());
```

Output:
```sql
    pragma foreign_keys = off;
    
    drop table if exists `parents`;
    drop table if exists `children`;
    
    create table `parents` (`id` integer not null primary key autoincrement);
    
    create table `children` (`id` integer not null primary key autoincrement, `name` varchar not null);
    create index `index_name_and_parent` on `children` (`name`, `parent_id`);                    -- ------ INDEX HERE
    
    alter table `children` add column `parent_id` integer null;               -- --------- COLUMN HERE
    create index `children_parent_id_index` on `children` (`parent_id`);
    
    pragma foreign_keys = on;
```

**Expected behavior**
Column should be created before the index.

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | `v12.16.1` |
| typescript | `3.9.6` |
| mikro-orm | `^3.6.15` |
| your-driver | `sqlite3` |

**Comment by B4nan** [comment]
This should be fixed in v4.

**Comment by B4nan** [comment]
Yes, it was fixed in v4 via #617
